### DISCUSSION TITLE
Number of steps for VITS recipe on VCTK training

### DISCUSSION THREAD

**Discussion by yurtmete** [coqui-ai/TTS#1734] [General Q&A]
Hi all, 

I am currently following your VCTK recipe for VITS model and I also tried your VITS model trained with VCTK dataset. The quality is very good considering the amount of data each speaker has. 

Based on the VITS paper, they trained the model for 800k steps with 64x4 (they use 4 GPUs) batch size. 

I was wondering, for many iterations have you trained the model. Based on the recipe, it seems that you use batch size 32 with single GPU. Did you wait until you hit 1000th epoch? 

Thank you in advance.

**Comment by yurtmete** [comment]
I downloaded the pretrained model using:
`$ wget https://coqui.gateway.scarf.sh/v0.6.1_models/tts_models--en--vctk--vits.zip`

Number of steps = 1000000
Number of epochs = 556
Batch size = 24 (this is an estimate based on the size of the dataset and the step, epoch sizes.)

Coqui guys can correct me if these are wrong.

**Reply by erogol** [reply]
why do you think it might be wrong?

**Reply by yurtmete** [reply]
I _roughly_ calculated the batch size, that is why I might need correction. 

As far as I know, the total number of samples are around 44000. Number of steps per epoch is around 1800 (# total_number_steps / # number of epochs). 

Then the batch size would be 44000 / 1800, which is around 24.

**Reply by erogol** [reply]
It is because of the incremental training where we started from a subset or LJspeech (I don't remember currently)

**Reply by yurtmete** [reply]
So the VITS VCTK model was not trained on full VCTK dataset from scratch?

**Reply by erogol** [reply]
AFAIR it started with LJSpeech and then fine-tuned with VCTK.

**Reply by yurtmete** [reply]
Understand, I am now trying to do the same, take your LJSpeech VITS model and fine tune it with VCTK dataset. This I think challenging in the sense that we go from single speaker to multi speaker. 

I followed your suggestions [here](https://tts.readthedocs.io/en/latest/finetuning.html). For the learning rates, what I have tried is that, I set the initial learning rate for the discriminator (since I do not have it in LJSpeech checkpoint) and for the generator do the following, if I am able to load the weights, I divide the initial learning rate by 10 to not to forget what LJSpeech model learnt, for the rest (like all the condition layers (there are no condition layers in LJSpeech model), text encoder (vocabulary is different), and speaker embedding) I set the initial learning since I will learn them from scratch. 

However, this did not work, and the model started to speak very fast. 

Do you suggest something else on finetuning LJSpeech model with VCTK voices? 

One final remark: **When finetuning, I am not using LJSpeech data at all**

**Reply by erogol** [reply]
Just train with LJSpeech once it converged, and continue with VCTK. You don't need to change anything I believe.

**Reply by yurtmete** [reply]
Not even reducing the learning rate?
